filetests/wasm/multi-val-reuse-ret-ptr-stack-slot.clif

```diff
@@ -0,0 +1,61 @@
+test legalizer
+target x86_64 haswell
+
+;; Test that we don't reuse `sret` stack slots for multiple calls. We could do
+;; this one day, but it would require some care to ensure that we don't have
+;; subsequent calls overwrite the results of previous calls.
```

Comment:
I'm assuming this is working as intended, but... isn't the max'ing happening in the `layout_stack` function all about trying to reuse the same slot? Could we also check the actual amount of stack pushed for the stack return space?